op-node/p2p/store/peer_ban_book.go

```diff
@@ -80,6 +76,8 @@ func (d *peerBanBook) SetPeerBanExpiration(id peer.ID, expirationTime time.Time)
 	if expirationTime == (time.Time{}) {
 		return d.book.deleteRecord(id)
 	}
+	d.book.Lock()
+	defer d.book.Unlock()
```

Comment:
As you noted in another comment, we probably also want to include the delete in the lock.